superset/db_engine_specs/databricks.py

```diff
@@ -72,19 +72,42 @@ def process(value: Any) -> str:
 
 def monkeypatch_dialect() -> None:
     """
-    Monkeypatch dialect to correctly escape single quotes.
-
-    The Databricks SQLAlchemy dialect we currently use does not escape single quotes
-    correctly -- it doubles the single quotes, instead of adding a backslash. The fixed
-    version requires SQLAlchemy 2.0, which is not yet available in Superset.
+    Monkeypatch dialect to correctly escape single quotes for Databricks.
+
+    The Databricks SQLAlchemy dialect (<3.0) incorrectly escapes single quotes by
+    doubling them ('O''Hara') instead of using backslash escaping ('O\'Hara'). The
+    fixed version requires SQLAlchemy>=2.0, which is not yet compatible with Superset.
+
+    The challenge: DatabricksDialect inherits from HiveDialect and accesses the
+    `colspecs` type mapping through its parent class rather than its own. This means:
+    - Direct assignment (`DatabricksDialect.colspecs[types.String] = ...`) works because
+    DatabricksDialect.colspecs points to the same dict object as HiveDialect.colspecs,
+    but then it affects all other dialects that inherit from HiveDialect, causing
+    incorrect escaping for them.
+    - Shallow copying then modifying `DatabricksDialect.colspecs` doesn't work
+    because it's never accessed.
```

Comment:
This seems more appropriate in the PR summary than as a comment in the code, since it mentions the previous attempt at fixing it. I'd rephrase it to something like:

> Since the `DatabricksDialect.colspecs` points to the base class (`HiveDialect.colspecs`) we can't patch it without affecting other Hive-based dialects. The solution is to introduce a dialect-aware string type so that the change applies only to Databricks.

Reply:
Done too!